app/models/manageiq/providers/embedded_terraform/automation_manager/job.rb

```diff
@@ -110,6 +106,13 @@ def stack_response
     @stack_response ||= Terraform::Runner::ResponseAsync.new(options[:terraform_stack_id])
   end
 
+  def decrypt_input_vars(input_vars)
+    input_vars
+      .dup
+      .fetch("extra_vars", {})
+      .transform_values { |val| val.kind_of?(String) ? ManageIQ::Password.try_decrypt(val) : val }
```

Comment:
This dup does not work as expected (it only dups the outer hash, but not the inner hash), however it's also not needed since transform_values will create a new hash anyway:

```
[1] pry(main)> input_vars = {"extra_vars" => {}}
[2] pry(main)> [input_vars.object_id, input_vars["extra_vars"].object_id]
=> [320, 340]
[3] pry(main)> iv2 = input_vars.dup
[4] pry(main)> [iv2.object_id, iv2["extra_vars"].object_id]
=> [360, 340]
[5] pry(main)> ev = iv2.fetch("extra_vars", {})
[6] pry(main)> ev.object_id
=> 340
[7] pry(main)> ev2 = ev.transform_values {}
[8] pry(main)> ev2.object_id
=> 400
```

Reply:
```
?>   def decrypt_input_vars(input_vars)
?>     result = input_vars.deep_dup
?>     result.fetch("extra_vars", {}).transform_values! { |val| val.kind_of?(String) ? ManageIQ::Password.try_decrypt(val) : val }
?>     result
>>   end
=> :decrypt_input_vars
>> input_vars = {"extra_vars" => {"foo" => "bar", "bar" => ManageIQ::Password.encrypt("baz")}}
=> {"extra_vars"=>{"foo"=>"bar", "bar"=>"v2:{8v956e2OHCGXgGW7Q7xxDw==}"}}
>> decrypt_input_vars(input_vars)
=> {"extra_vars"=>{"foo"=>"bar", "bar"=>"baz"}}
>> input_vars
=> {"extra_vars"=>{"foo"=>"bar", "bar"=>"v2:{8v956e2OHCGXgGW7Q7xxDw==}"}}
```

I want to avoid modifying the original options and saving a decrypted value, went with deep_dup